src/jit/RegisterAlloc.cpp

```diff
@@ -87,26 +98,43 @@ class RegisterSet {
 
 class RegisterFile {
 public:
-    RegisterFile(uint32_t numberOfIntegerScratchRegs, uint32_t numberOfIntegerSavedRegs,
-                 uint32_t numberOfFloatScratchRegs, uint32_t numberOfFloatSavedRegs)
+    RegisterFile(uint32_t numberOfIntegerScratchRegs, uint32_t numberOfIntegerSavedRegs)
         : m_integerSet(numberOfIntegerScratchRegs, numberOfIntegerSavedRegs, true)
-        , m_floatSet(numberOfFloatScratchRegs, numberOfFloatSavedRegs, false)
+        , m_floatSet(SLJIT_NUMBER_OF_SCRATCH_FLOAT_REGISTERS, SLJIT_NUMBER_OF_SAVED_FLOAT_REGISTERS, false)
+#if (defined SLJIT_SEPARATE_VECTOR_REGISTERS && SLJIT_SEPARATE_VECTOR_REGISTERS)
+#if (defined SLJIT_CONFIG_RISCV && SLJIT_CONFIG_RISCV)
+        , m_vectorSet(SLJIT_NUMBER_OF_SCRATCH_VECTOR_REGISTERS - 1, SLJIT_NUMBER_OF_SAVED_VECTOR_REGISTERS, false)
+#else /* !SLJIT_CONFIG_RISCV */
+        , m_vectorSet(SLJIT_NUMBER_OF_SCRATCH_VECTOR_REGISTERS, SLJIT_NUMBER_OF_SAVED_VECTOR_REGISTERS, false)
+#endif /* SLJIT_CONFIG_RISCV */
+#endif /* SLJIT_SEPARATE_VECTOR_REGISTERS */
     {
     }
 
     RegisterSet& integerSet() { return m_integerSet; }
     RegisterSet& floatSet() { return m_floatSet; }
-
+    // clang-format off
+#if (defined SLJIT_SEPARATE_VECTOR_REGISTERS && SLJIT_SEPARATE_VECTOR_REGISTERS)
+    RegisterSet& vectorSet() { return m_vectorSet; }
+#endif /* SLJIT_SEPARATE_VECTOR_REGISTERS */
+    // clang-format on
     uint8_t toCPUIntegerReg(uint8_t reg)
     {
-        return m_integerSet.toCPUReg(reg);
+        return m_integerSet.toCPUReg(reg, SLJIT_R0, SLJIT_S2);
     }
 
     uint8_t toCPUFloatReg(uint8_t reg)
     {
-        return m_floatSet.toCPUReg(reg);
+        return m_floatSet.toCPUReg(reg, SLJIT_FR0, SLJIT_FS0);
     }
 
+#if (defined SLJIT_SEPARATE_VECTOR_REGISTERS && SLJIT_SEPARATE_VECTOR_REGISTERS)
+    uint8_t toCPUVectorReg(uint8_t reg)
+    {
+        return m_vectorSet.toCPUReg(reg, SLJIT_VR1, SLJIT_VS0);
```

Comment:
SLJIT_VR1 is riscv only thing. Maybe you could introduce a macro for the first register, and use it everywhere. SLJIT_VR1 on riscv, SLJIT_VR0 everywhere else.